sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingDeduplicationSuite.scala

```diff
@@ -484,6 +486,52 @@ class StreamingDeduplicationSuite extends StateStoreMetricsTest {
       CheckLastBatch(("c", 9, "c"))
     )
   }
+
+  test("collation aware deduplication") {
+    val inputData = MemoryStream[(String, Int)]
+    val result = inputData.toDF()
+      .select(col("_1")
+        .try_cast(StringType("UNICODE")).as("str"),
```

Comment:
While we are here, I see UNICODE is binary equality but non-binary ordering. Does this still ensure that we can put this into RocksDB which key is binary sorted and find the key group based on prefix of key including this column?

E.g. Say we have two columns, dept (String with UNICODE collation), session start (timestamp) as grouping key, and want to scan all grouping keys which are having dept as 'dept1'. This is required for several operations like session window aggregation.

My gut feeling is yes, but I would like to double confirm.

Reply:
In theory, it should work, but we need to test this as well.
binary ordering means that you can use binary representation to check which string alphabetically comes first. But if we care only about equality (which is usually used in "groupings" and joins), binary equality is only important.

I will follow up with additional testing.

Reply:
Thanks, using UNICODE in grouping key of session window aggregation should cover it.